recipes/mingw-w64/all/conandata.yml

```diff
@@ -1,18 +1,70 @@
+            sjlj:
+              url: "https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win64/Personal%20Builds/mingw-builds/8.1.0/threads-win32/sjlj/x86_64-8.1.0-release-win32-sjlj-rt_v6-rev0.7z"
+              sha256: "e8c65ddc655534b0330f66f7b480565621e8617cda9937d76ba141a22bf3b2fa"
+  "8.0.2":
+    url:
+      Linux:
+        x86_64:
+          mingw-w64:
+            url: "https://sourceforge.net/projects/mingw-w64/files/mingw-w64/mingw-w64-release/mingw-w64-v8.0.2.tar.bz2"
+            sha256: "f00cf50951867a356d3dc0dcc7a9a9b422972302e23d54a33fc05ee7f73eee4d"
+          binutils:
+            url: "http://ftp.gnu.org/gnu/binutils/binutils-2.35.2.tar.bz2"
+            sha256: "cfa7644dbecf4591e136eb407c1c1da16578bd2b03f0c2e8acdceba194bb9d61"
+          gcc:
+            url: "http://ftp.gnu.org/gnu/gcc/gcc-10.3.0/gcc-10.3.0.tar.xz"
+            sha256: "64f404c1a650f27fc33da242e1f2df54952e3963a49e06e73f6940f3223ac344"
+          gmp:
+            url: "https://ftp.gnu.org/gnu/gmp/gmp-6.2.1.tar.xz"
+            sha256: "fd4829912cddd12f84181c3451cc752be224643e87fac497b69edddadc49b4f2"
+          mpfr:
+            url: "https://ftp.gnu.org/gnu/mpfr/mpfr-4.1.0.tar.xz"
+            sha256: "0c98a3f1732ff6ca4ea690552079da9c597872d30e96ec28414ee23c95558a7f"
+          mpc:
+            url: "https://ftp.gnu.org/gnu/mpc/mpc-1.2.1.tar.gz"
+            sha256: "17503d2c395dfcf106b622dc142683c1199431d095367c6aacba6eec30340459"
+          isl:
+            url: "http://isl.gforge.inria.fr/isl-0.23.tar.bz2"
+            sha256: "c58922c14ae7d0791a77932f377840890f19bc486b653fa64eba7f1026fb214d"
+  "9.0.0":
+    url:
+      Linux:
+        x86_64:
+          mingw-w64:
+            url: "https://sourceforge.net/projects/mingw-w64/files/mingw-w64/mingw-w64-release/mingw-w64-v9.0.0.tar.bz2"
+            sha256: "f00cf50951867a356d3dc0dcc7a9a9b422972302e23d54a33fc05ee7f73eee4d"
+          binutils:
+            url: "http://ftp.gnu.org/gnu/binutils/binutils-2.35.2.tar.bz2"
+            sha256: "cfa7644dbecf4591e136eb407c1c1da16578bd2b03f0c2e8acdceba194bb9d61"
+          gcc:
+            url: "http://ftp.gnu.org/gnu/gcc/gcc-10.3.0/gcc-10.3.0.tar.xz"
+            sha256: "64f404c1a650f27fc33da242e1f2df54952e3963a49e06e73f6940f3223ac344"
+          gmp:
+            url: "https://ftp.gnu.org/gnu/gmp/gmp-6.2.1.tar.xz"
+            sha256: "fd4829912cddd12f84181c3451cc752be224643e87fac497b69edddadc49b4f2"
+          mpfr:
+            url: "https://ftp.gnu.org/gnu/mpfr/mpfr-4.1.0.tar.xz"
+            sha256: "0c98a3f1732ff6ca4ea690552079da9c597872d30e96ec28414ee23c95558a7f"
+          mpc:
+            url: "https://ftp.gnu.org/gnu/mpc/mpc-1.2.1.tar.gz"
+            sha256: "17503d2c395dfcf106b622dc142683c1199431d095367c6aacba6eec30340459"
+          isl:
+            url: "http://isl.gforge.inria.fr/isl-0.23.tar.bz2"
+            sha256: "c58922c14ae7d0791a77932f377840890f19bc486b653fa64eba7f1026fb214d"
```

Comment:
We have conan packages for these 4 projects.
Perhaps we can re-use those?

Reply:
Good point, I will check. 
isl is not even used, so I will remove that one.

Reply:
I tried to use the gmp, mpfr and mpc conan packages, but somehow these are not working.
See my commented-out code.

I always get the error
```
checking for ld that supports -Wl,--gc-sections... configure: error: Link tests are not allowed after GCC_NO_EXECUTABLES.
Makefile:11275: recipe for target 'configure-target-libstdc++-v3' failed
```

which means that somewhere before gcc already failed to compile itself, and therefore it cannot generate executables. If I just use the build directly (i.e., download sources for gmp, mpc, mpfr, and build them), it works. Did not yet find out why.

This has to wait for now, cannot have a deeper look into it, unfortunately.

Reply:
The reason it fails might be because `AutotoolsBuildEnvironment` can be too intrusive.
It automatically adds the libraries of all dependencies to the `LIBS` environment variable.
Remove them by doing the following to every `autotools` object
```python
autotools.libs = []
```
and/or
```python
autotools.defines = []
```

Reply:
I think the `AutotoolsBuildEnvironment` is not the problem here, I.e., the current version in this MR works.
That is directly downloading gmp/mpc/mpfr and then using the `AutotoolsBuildEnvironment` to build. So that works fine.

The issue comes in if I use the conan packages of gmp/mpc/mpfr as dependencies. My wild guess is that the conan package versions use som specific settings which are not compatible, OR the mingw build expects those files in the sysroot (i.e., package_folder).

Reply:
I did various tests trying to use the pre-existing conan packages for gmp, mpc, and mpfr, but I always end up getting
```
checking for ld version... 23502
checking for ld that supports -Wl,--gc-sections... configure: error: Link tests are not allowed after GCC_NO_EXECUTABLES.
Makefile:11275: recipe for target 'configure-target-libstdc++-v3' failed
```
when building libgcc.

I tried using `RunEnvironment()` and setting `LD_LIBRARY_PATH` manually, but no success.

If I build from source within the mingw conanfile.py, it works. So there must be some issue with the conan packages of gmp, mpc or mpfr, or an issue to get their environment into mingw cross compile.

I am out of ideas.

Reply:
Seems to be solved now, yayy. Wrong compiler was used.